src/object_store.hpp

```diff
@@ -18,9 +18,11 @@ namespace ddwaf
 class object_store
 {
 public:
-    explicit object_store(const manifest& m): manifest_(m) {}
+    explicit object_store(const manifest& m,
+        ddwaf_object_free_fn free_fn = ddwaf_object_free);
+    ~object_store();
 
-    bool insert(const ddwaf_object &input);
+    bool insert(ddwaf_object &input);
```

Comment:
The reason is that `ddwaf_object_free` takes a non-const object, although I guess it doesn't need to and also we can still take a const reference since we're simply copying the contents anyway.